### PR TITLE
EZP-29083: Error after trying to delete role assignment

### PR DESCRIPTION
**Pull request by ViniTou**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-29083](https://jira.ez.no/browse/EZP-29083)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | `7.2` 
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

Cache related to role-assignment wasn't invalidated when editing policies list.

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Write tests
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/Core/Persistence/Cache/UserHandler.php [hunk 1]
```diff
@@ -24,6 +24,8 @@ class UserHandler extends AbstractHandler implements UserHandlerInterface
 {
     /**
      * {@inheritdoc}
+     *
+     * @throws \Psr\Cache\InvalidArgumentException
```
Comment (andrerom) [inline eZ/Publish/Core/Persistence/Cache/UserHandler.php]: invalid edits, not for this PR, but if you are truly seeing this exception then we need to catch it inside the cache impl here and not let it leak true.

**Comment by andrerom** [comment]
Usually I would say yes.
_But_, role assignment does not contain roles, just id reference to the role.

So _which_ cache and _in exactly which_ case does the cache becomes stale here?

**Comment by ViniTou** [comment]
@andrerom 
Not sure if this answers your question, but:

>So which cache
`ez-role-assignment-`

>and in exactly which case
There is user or group assigned to role. Whenever you change policies attached to that role (add new one, edit existing limitations or removing existing policy) there is an update of id in `ezuser_role`. But that change is not reflected when loading role assignments.

File: eZ/Publish/Core/Persistence/Cache/UserHandler.php (hunk 1 above)
Comment (ViniTou) [reply eZ/Publish/Core/Persistence/Cache/UserHandler.php]: I never saw them per se, that was just catch up by static code analysis. Should I then revert those?

**Comment by ViniTou** [comment]
@alongosz 
@andrerom 
Feel free to take another look :)

File: eZ/Publish/Core/Persistence/Cache/UserHandler.php [hunk 2]
```diff
@@ -306,6 +306,7 @@ public function loadRoleAssignmentsByRoleId($roleId)
         $cacheItem->set($roleAssignments);
         $cacheTags = ['role-assignment-role-list-' . $roleId];
         foreach ($roleAssignments as $roleAssignment) {
+            $cacheTags[] = 'role-' . $roleId; /* Role update (policies) changes role assignment id */
```
Comment (andrerom) [inline eZ/Publish/Core/Persistence/Cache/UserHandler.php]: can move this just before the foreach, it's the same for all items.

**Comment by lserwatka** [comment]
@ViniTou Yes!

**Comment by lserwatka** [comment]
@ViniTou could you merge it up to master?

**Comment by ViniTou** [comment]
@lserwatka 
merged

**Pull request merged**